### PR TITLE
Fix test_ceph_pg_log_dups_trim. Replace oc cp calls (Updated)

### PR DESCRIPTION
**Pull request by DanielOsypenko**
Fixes https://github.com/red-hat-storage/ocs-ci/issues/7140
Commit fixes the test tests/manage/z_cluster/test_ceph_pg_log_dups_trim.py::TestCephPgLogDupsTrimming::test_ceph_pg_log_dups_trim
and 'oc cp' to the pod references.

Due to implemented https://url.corp.redhat.com/RHSTOR-3411 task 'tar', 'yum' utilities were remove from ceph pods to trim an image size.
oc cp command depends on 'tar' utility, see https://linuxhint.com/use-kubectl-cp-command/ and oc cp --help

!!!Important Note!!!
Requires that the 'tar' binary is present in your container
image. If 'tar' is not present, 'oc cp' will fail.

therefore it fails, hence func copy_to_pod_cat and copy_from_pod_oc_exec added to the framework.

Copy to rook-ceph-osd with oc rsync also fails with error:

WARNING: cannot use rsync: rsync not available in container
WARNING: cannot use tar: tar not available in container
error: No available strategies to copy.

~~Minor improvements: exec_cmd - parameter added 'use_shell=False' ; ZombieProcess's, NoSuchProcess's stack trace print removed~~

High Priority: Test has longstanding failure history, has a recently opened bug and should run to close bug and trace failures.

merge conflict resolved, all commits are signed

### CODE REVIEW

File: ocs_ci/utility/memory.py [hunk 1]
```diff
@@ -73,10 +73,13 @@ def _rec_memory(proc: Process):
...
         )
+    # ZombieProcess's, NoSuchProcess's come too often within a test run,
+    # we're polling each process once per 3 sec. ZombieProcess and NoSuchProcess
+    # appear due to concurrency. Failed polls are not valuable information
     except ZombieProcess:
-        log.exception("Cannot access ZombieProcess. Recording skipped")
+        pass
     except NoSuchProcess:
-        log.exception("Process is unavailable. Recording skipped")
+        pass
```
Comment (dahorak) [inline ocs_ci/utility/memory.py:82]: I would propose to move this fix to separated PR as it is not related to the test `test_ceph_pg_log_dups_trim` and also we might want to backport it to the release branches (if it is a fix for Issue https://github.com/red-hat-storage/ocs-ci/issues/7171).

Comment (DanielOsypenko) [reply ocs_ci/utility/memory.py]: reverted the change

Comment (DanielOsypenko) [reply ocs_ci/utility/memory.py]: @dahorak extracted to https://github.com/red-hat-storage/ocs-ci/pull/7181

**Review by shivamdurgbuns — APPROVED** [review]
LGTM

**Review by yitzhak12 — APPROVED** [review]

File: ocs_ci/ocs/resources/pod.py [hunk 2]
```diff
@@ -194,7 +197,7 @@ def exec_s3_cmd_on_pod(self, command, mcg_obj=None):
             else None,
         )
 
-    def copy_to_pod(self, src_path, target_path, container=None):
+    def copy_to_pod_rsync(self, src_path, target_path, container=None):
```
Comment (PrasadDesala) [inline ocs_ci/ocs/resources/pod.py:200]: I added a comment on this in the old PR and it looks like it was not addressed. adding it here again.

```You have renamed the existing function, please make sure that there is no regression with this change. For instance this function is used by javasdk_pod_fixture
```

File: ocs_ci/ocs/resources/pod.py [hunk 3]
```diff
@@ -212,6 +215,60 @@ def copy_to_pod(self, src_path, target_path, container=None):
...
+            timeout (int): timeout in seconds
+                until size of the local file will not reach the initial file size.
+                2000 lines is a maximum chunk size tested successfully
+            chunk_size (int): file will be copied by chunks, by number of lines
+        """
+        file_size = int(self.exec_cmd_on_pod(f"stat -c %s {src_path}"))
+
+        timestamp_end = datetime.now() + timedelta(seconds=timeout)
+
+        x = 1
```
Comment (PrasadDesala) [inline ocs_ci/ocs/resources/pod.py:252]: can we give some meaningful var name?

File: ocs_ci/ocs/resources/pod.py (hunk 2 above)
Comment (DanielOsypenko) [reply ocs_ci/ocs/resources/pod.py:200]: discussed, ++ done

File: ocs_ci/ocs/resources/pod.py (hunk 3 above)
Comment (DanielOsypenko) [reply ocs_ci/ocs/resources/pod.py]: done x -> start_line

**Review by PrasadDesala — APPROVED** [review]

**Review by shivamdurgbuns — APPROVED** [review]

**Review by petr-balogh — APPROVED** [review]

**Pull request merged**